modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/ColumnStatisticsCollector.java

```diff
@@ -136,6 +137,13 @@ public ColumnStatistics finish() {
 
         int averageSize = averageSize(size, total, nullsCnt);
 
+        TypeInfo colTypeInfo = col.getType();
+
+        // Avoid serializing complex types.
+        if (colTypeInfo == TypeInfo.TYPE_ARRAY || colTypeInfo == TypeInfo.TYPE_ENUM_UNDEFINED
```

Comment:
why we are collecting them then?

Reply:
We still can use some its statistics, null counter, for example.

Reply:
I mean why we are still collecting part with min|max? It takes time to compare these values, but at the end we just discard work. May be we need to skip collecting min|max at all for those types

Reply:
Got it. Fixed.